File: hotel-service/hotel-service.API/common/basemodel/response/BaseResponse.cs
namespace hotel_service.API.basemodel.response;

public class BaseResponse<T>
{
    public bool IsSuccess { get; set; }
    public T Data { get; set; }
    public string ExecutionDate { get; set; }
    public string Message { get; set; }
    public ErrorResponse Error { get; set; }

    public BaseResponse(bool isSuccess, T data)
    {
        IsSuccess = isSuccess;
        Data = data;
        ExecutionDate = DateTimeOffset.Now.ToString("O");
    }

    public BaseResponse(bool isSuccess, string message)
    {
        IsSuccess = isSuccess;
        Message = message;
        ExecutionDate = DateTimeOffset.Now.ToString("O");
    }

    public BaseResponse(bool isSuccess, ErrorResponse error)
    {
        IsSuccess = isSuccess;
        Error = error;
        ExecutionDate = DateTimeOffset.Now.ToString("O");
    }
    
}

File: booking-service/src/main/java/com/example/bookingservice/common/base/response/BaseResponse.java
package com.example.bookingservice.common.base.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.time.ZonedDateTime;
import java.time.format.DateTimeFormatter;

@Getter
@Setter
@Builder
@AllArgsConstructor
public class BaseResponse<T> {

    private boolean isSuccess;
    private T data;
    private String executionDate;
    private String message;
    private Error error;

    private BaseResponse() {}

    public BaseResponse(boolean isSuccess, T data) {
        this.isSuccess = isSuccess;
        this.data = data;
        this.executionDate = ZonedDateTime.now().format(DateTimeFormatter.ISO_LOCAL_DATE_TIME);
    }

    public BaseResponse(boolean isSuccess, String message) {
        this.isSuccess = isSuccess;
        this.message = message;
        this.executionDate = ZonedDateTime.now().format(DateTimeFormatter.ISO_LOCAL_DATE_TIME);
    }

    public BaseResponse(boolean isSuccess, Error error) {
        this.isSuccess = isSuccess;
        this.error = error;
        this.executionDate = ZonedDateTime.now().format(DateTimeFormatter.ISO_LOCAL_DATE_TIME);
    }

}
